plugins/thelio-io/fu-thelio-io-device.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2019 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-io-channel.h"
+#include "fu-thelio-io-device.h"
+
+struct _FuThelioIoDevice {
+	FuUsbDevice			 parent_instance;
+};
+
+G_DEFINE_TYPE (FuThelioIoDevice, fu_thelio_io_device, FU_TYPE_USB_DEVICE)
+
+static gboolean
+fu_thelio_io_device_probe (FuDevice *device, GError **error)
+{
+	gssize read_size;
+	g_autofree gchar *fn = NULL;
+	g_autoptr(GUdevDevice) udev_device = NULL;
+	g_autoptr(GFile) file = NULL;
+	g_autoptr(GInputStream) istr = NULL;
+	guint8 buf[4096] = { 0 };
+
+	fu_device_add_instance_id (device, "USB\\VID_03EB&PID_2FF4");
+
+	/* convert GUsbDevice to GUdevDevice */
+	udev_device = fu_usb_device_find_udev_device (FU_USB_DEVICE (device), error);
+	if (udev_device == NULL)
+		return FALSE;
+
+	fn = g_build_filename (g_udev_device_get_sysfs_path (udev_device), "revision", NULL);
+	file = g_file_new_for_path (fn);
+	istr = G_INPUT_STREAM (g_file_read (file, NULL, error));
+	if (istr == NULL)
+		return FALSE;
+
+	read_size = g_input_stream_read (istr, buf, sizeof(buf) - 1, NULL, error);
+	if (read_size < 0) {
+		fu_device_set_version (device, "0.0.0", FWUPD_VERSION_FORMAT_TRIPLET);
+	} else {
+		buf[read_size] = 0;
+		fu_device_set_version (device, (const gchar *) buf, FWUPD_VERSION_FORMAT_TRIPLET);
+	}
```

Comment:
Technically isn't sending buf the same in all cases? So you don't need a different case of calling set version in the < 0 case.

If it's less than zero you should be sending a buf that was initialized to zero above, no?